tests/subsys/settings_commit_prio/src/main.c

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (c) 2024 Laczen
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <errno.h>
+#include <zephyr/settings/settings.h>
+#include <zephyr/ztest.h>
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(test);
+
+static int prio;
+
+int commit0(void)
+{
+	LOG_INF("%s Called", __func__);
+	zassert_equal(prio, 0, "Bad commit order");
+	prio++;
+	return 0;
+}
+
+int commit1(void)
+{
+	LOG_INF("%s Called", __func__);
+	zassert_equal(prio, 1, "Bad commit order");
+	prio++;
+	return 0;
+}
+
+int commit2(void)
+{
+	LOG_INF("%s Called", __func__);
+	zassert_equal(prio, 2, "Bad commit order");
+	prio++;
+	return 0;
+}
+
+int commit3(void)
+{
+	LOG_INF("%s Called", __func__);
+	zassert_equal(prio, 3, "Bad commit order");
+	prio++;
+	return 0;
+}
+
+int commit5(void)
+{
+	LOG_INF("%s Called", __func__);
+	zassert_equal(prio, 0, "Bad commit order");
+	return 0;
+}
+
+SETTINGS_STATIC_HANDLER_DEFINE_WITH_CPRIO(h0, "h0", NULL, NULL, commit0, NULL, 0);
+SETTINGS_STATIC_HANDLER_DEFINE_WITH_CPRIO(h2, "h2", NULL, NULL, commit2, NULL, 2);
+
+static struct settings_handler h1 = {
+	.name = "h1",
+	.h_commit = commit1,
+};
+
+static struct settings_handler h3 = {
+	.name = "h3",
+	.h_commit = commit3,
+};
+
+SETTINGS_STATIC_HANDLER_DEFINE_WITH_CPRIO(h5, "h5", NULL, NULL, commit5, NULL, -1);
```

Comment:
The order of these is a bit confusing as the commits are ordered by priority (except for `commit5`), but the order here in terms of priority is

0
2
1
3
5

It works, but is a slightly confusing to read :) Doesn't need a change though if you think this works the best

Reply:
The idea was that h5 was added later on in a development process and needed to be highest priority. If needed I can add a remark for it.

The definition order is 0,2,1,3,5 (intentionally mixing static and dynamic handlers). The resulting commit order is 5, 0, 1, 2, 3.